modules/nf-core/openms/idmerger/tests/main.nf.test

```diff
@@ -0,0 +1,66 @@
+nextflow_process {
+
+    name "Test Process OPENMS_IDMERGER"
+    script "../main.nf"
+    process "OPENMS_IDMERGER"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "openms"
+    tag "openms/idmerger"
+
+    test("proteomics - openms - idxml") {
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                            [ id:'test'], // meta map
+                            [ file(params.test_data['proteomics']['openms']['idxml1'], checkIfExists: true),
+                              file(params.test_data['proteomics']['openms']['idxml2'], checkIfExists: true)]
+                        ])
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match()},
+                
+                // check single line substrings in output idxml file
+                { assert path(process.out.idxml.get(0).get(1)).readLines().any { it.contains("</PeptideHit>") }},
```

Comment:
in the old pytest this worked:
path: output/openms/test.idXML
md5sum: a3b69bbe0e4818cd37d8c6faccbb2090
Isn't it the case anymore?

Reply:
I had an issue that I would always get one of two different md5sums for the idxml (`a3b...` or `abbb...`)
I could not find a pattern behind it. Sometimes I would get the same md5 for several runs and then the other one again.

Usually when updating the snapshot, it stored `abbb...` as md5, then, when rerunning the test it sometimes passed and sometimes failed with  the `a3b...` sum.

Reply:
You'll have to sort the files first, e.g.:
file(process.out.idxml.get(0).get(1)).list().sort()
Example module: https://github.com/nf-core/modules/blob/master/modules/nf-core/bclconvert/tests/main.nf.test

Reply:
does this sort the output files or the lines within an output file?

Reply:
The output file. Then you'll figure out which one [0] or [1] is the one you want to md5sum (a3b or abbb) and proceed with that!

Reply:
thanks this seems like exactly what I was looking for. however I did not get it to run yet.
`{ assert snapshot(file(process.out.idxml.get(0).get(1)).list().sort()).match()}`
throws an error
`java.lang.NullPointerException: Cannot invoke method sort() on null object `

and 
`{ assert snapshot(file(process.out.idxml.get(0).get(1)).readLines().sort()).match()}`
does not md5sum the sorted lines but stores them in the snapshot

any thoghts?

Reply:
Sort won't work on the readLines. But you won't need the readLines once if we get the other to work. Maybe the example wasn't quite on spot. Try something like this? (taken from: https://github.com/nf-core/modules/blob/master/modules/nf-core/star/genomegenerate/tests/main.nf.test):

assert snapshot(file(process.out.idxml[0][1]).listFiles().collect { it.getName() }.sort().toString()).match("name_of_your_file_here")

For more examples that could help in your case, search repo wide for .sort() in *main.nf.test in the repo ;)

Reply:
hm this did not work for me, and it seems to me, that this would rather sort the output filenames instead of the lines, right?
I did not find any similar use cases (but admittedly did not search insanely throughout)
should I further pursue this? otherwise I would leave it with the four substring checks.